### ISSUE TITLE
useOutsideClick ignores right-click

### ISSUE THREAD

**Issue by shanberg** [chakra-ui/chakra-ui#5829]
### Description

When I use `useOutsideClick`, I expect it to respond to all types of click, not just regular click.

### Link to Reproduction

https://codesandbox.io/s/useoutsideclick-ignores-right-click-yit1y5?file=/src/App.js

### Chakra UI Version

1.8.7

**Comment by TylerAPfledderer** [comment]
This might be an accessibility consideration to account for folks who may use the right-click as a primary: left-handed folks or those with motor issues in the first finger.

Currently any mouse event button not the primary (0) is ignored in the hook.

Based on what I stated above, probably allow the right-click by default and turn off that or left-click via a prop?

**Comment by TylerAPfledderer** [comment]
@shanberg circled back to this. 

My above argument does not hold because some users might us a mouse that only has a singular button, or they have the ability to configure the button mapping via OS or mouse driver settings (current OS settings should have this available).

I guess the question at this point is why the need for a different click event for this hook? What use cases are you running into that requires a click that is not the primary?

**Comment by shanberg** [comment]
@TylerAPfledderer For context menus.

I think my repro is still relevant: https://codesandbox.io/s/useoutsideclick-ignores-right-click-yit1y5?file=/src/App.js I expect that any interaction (including right-click) outside of a context menu should close any open menus.

I don't know what the best solution is, but the name of the hook seems ambiguous or misleading if it’s only intended for to handle primary clicks.

Some concepts I think might address my case
- An abstracted version of the hook e.g. useOutsideInteraction, with paramaters to define the interactions to watch for.
- Leave the existing defaults for useOutsideClick, but allow configuration.
- A Chakra context menu component/hook with the behavior built-in.
- Update then name or documentation of useOutsideClick to clarify it’s meant only for primary clicks.

**Comment by stephanepx** [comment]
I think it should work if we replace this line https://github.com/chakra-ui/chakra-ui/blob/main/packages/hooks/use-outside-click/src/index.ts#L79 by this : 

```typescript
  if (event.button > 0 && event.which !== 3) return false
```

**Comment by segunadebayo** [comment]
This has already been fixed in one of the recent PRs. We'll release an update shortly.

If the issue persists after the release, I'll re-open this issue.
